dialogs/ActionsBar.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <QDialog>
+#include <QPixmap>
+#include <QAction>
+#include <QToolBar>
+
+#include <functional>
+
+namespace ui
+{
+	// This exists purely for the zoo application
+	class CActionsBarDialog : public QDialog
+	{
+		Q_OBJECT
+	public:
+		CActionsBarDialog( QWidget *pParent );
+
+	private:
+		class CActionsBar *m_pHActionsBar;
+		class CActionsBar *m_pVActionsBar;
+	};
+
+	//-----------------------------------------------------------------------------------------//
+
+	// Wrapper for QToolBar to make stuff a little less verbose
+	class CActionsBar : public QWidget
+	{
+		Q_OBJECT
+
+	public:
+		CActionsBar( QWidget *pParent, Qt::Orientation orientation = Qt::Horizontal );
+
+		// Todo: Is this name really clear? QWidget already has addAction
+		// so we dont really have any alternatives
+		QAction *addButton( const QString &pIconPath, const QString &pToolTip, const bool bCheckable, const std::function<void()> &func );
```

Comment:
Not a fan of using `std::function` here. We don't use it in our codebase primarily because of the added compile time. 
Ideally the function would be a template parameter. You could have a private version of `addButton` that returns the action (without connecting func), and then implement the templated version in the header, which will connect `func`.

```cpp
template<class Func>
QAction *addButton( const QString &pIconPath, const QString &pToolTip, const bool bCheckable, Func func ) {
    auto* pAction = addButton(pIconPath, pToolTip, bCheckable);
    connect(pAction, &QAction::somethingIdk, std::move(func));
}

template<class Object, class Func>
QAction *addButton( const QString &pIconPath, const QString &pToolTip, const bool bCheckable, const Object& obj, Func func ) {
    auto* pAction = addButton(pIconPath, pToolTip, bCheckable);
    connect(pAction, &QAction::somethingIdk, obj, std::move(func));
}

```

`bCheckable` doesn't really need to be const since it's just pass by value. I suppose it couldn't hurt though..

Reply:
Originally I was going to use a template, but I spoke with Craftable and he suggested `std::function`, I can easily change it though!